core/network-libp2p/src/custom_proto/behaviour.rs

```diff
@@ -372,10 +372,33 @@ impl<TMessage, TSubstream> CustomProtos<TMessage, TSubstream> {
 				.count();
 			self.max_outgoing_connections - num_outgoing_connections
 		};
-
 		trace!(target: "sub-libp2p", "Connect-to-nodes round; attempting to fill {:?} slots",
 			num_to_open);
 
+		// We first try to enable existing connections.
+		for peer_id in &self.connected_peers {
+			if num_to_open == 0 {
+				break
+			}
+
+			if self.enabled_peers.contains_key(peer_id) {
+				continue;
+			}
+
+			if let Some((_, expire)) = self.banned_peers.iter().find(|(p, _)| p == peer_id) {
```

Comment:
Just out of curiosity, do we hold  connection to a peer that is banned?

Reply:
Yes. Banned peers can still perform Kademlia queries.
The connection should however close after a few seconds because it is marked as "useless".